src/libplctag/Tag.cs

```diff
@@ -150,7 +150,11 @@ public async Task ReadAsync(CancellationToken token = default)
 
             do
             {
-                token.ThrowIfCancellationRequested();
+                if(token.IsCancellationRequested)
+                {
+                    Abort();
+                    token.ThrowIfCancellationRequested();
+                }
                 await Task.Delay(ASYNC_STATUS_POLL_INTERVAL);
```

Comment:
I know it's only a 2 msec poll interval, but we should probably stick the token in the `Delay` as long as we have it...

Reply:
Done